internal/osbuildexecutor/runner-aws-ec2_test.go

```diff
@@ -0,0 +1,135 @@
+package osbuildexecutor
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/osbuild/images/pkg/osbuild"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWaitForSI(t *testing.T) {
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel()
+	require.False(t, waitForSI(ctx, server.URL))
+
+	server.Start()
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel2()
+	require.True(t, waitForSI(ctx2, server.URL))
+}
+
+func TestWriteInputArchive(t *testing.T) {
+	cacheDir, err := os.MkdirTemp("", "")
```

Comment:
(nitpick) you could use `t.TempDir()` here which will not require error handling or cleanup :) (and in some similar places)